src/shm/provider/shared_memory_provider_backend.rs

```diff
@@ -0,0 +1,104 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+use std::fmt::Debug;
+
+use libc::c_void;
+use zenoh::core::Result;
+use zenoh::shm::{ChunkAllocResult, ChunkDescriptor, MemoryLayout, SharedMemoryProviderBackend};
+use zenoh_util::core::zerror;
+
+use crate::context::DroppableContext;
+use crate::transmute::{TransmuteIntoHandle, TransmuteRef};
+use crate::{z_loaned_memory_layout_t, z_owned_chunk_alloc_result_t, z_owned_memory_layout_t};
+
+use super::chunk::z_chunk_descriptor_t;
+
+/// A callbacks for SharedMemoryProviderBackend
+#[derive(Debug)]
+#[repr(C)]
+pub struct zc_shared_memory_provider_backend_callbacks_t {
+    alloc_fn: unsafe extern "C" fn(
+        *mut c_void,
+        &z_loaned_memory_layout_t,
+        *mut z_owned_chunk_alloc_result_t,
+    ),
+    free_fn: unsafe extern "C" fn(*mut c_void, *const z_chunk_descriptor_t),
```

Comment:
Could you provide names to function arguments (to better reflect their meaning) ? Also,  If the first c_void* is meant to represent some context, we tend to pass it as the last argument in the function call.